ui/components/src/Legend/Legend.tsx

```diff
@@ -23,25 +23,35 @@ export interface LegendProps {
   options: LegendOptions;
 }
 
+// When the number of items to display is above this number, it is likely to
+// cause performance issues in the browser. The legend will be displayed in a
+// format (list) that allows for virtualization to minimize the performance impact.
+// Set this number based on testing, but it may need to be tuned a bit on the
+// future as people test this out on different machines.
+const NEED_VIRTUALIZATION_LIMIT = 500;
+
 export function Legend({ width, height, options, data }: LegendProps) {
   if (options.position === 'Right') {
     return (
       <Box
         sx={{
           width: width,
-          height: '100%',
+          height: height,
```

Comment:
Using the specified height is probably more "correct" from an expectation perspective AND, more importantly, enables us to use the height to do calculations for the virtualization. Otherwise, we'd need something like autosizer, which adds even more complexity to this already complex code.